guides/source/association_basics.md

```diff
@@ -1301,10 +1301,40 @@ By convention, Rails assumes that the column used to hold the primary key of thi
 
 The `:source` option specifies the source association name for a `has_one :through` association.
 
+```ruby
+class Supplier < ApplicationRecord
+  has_one :account
+  has_one :history, through: :account, source: :account_history
+end
+
+class Account < ApplicationRecord
+  belongs_to :supplier
+  has_one :account_history
+end
+
+class AccountHistory < ApplicationRecord
+  belongs_to :account
+end
+```
+
 ##### `:source_type`
 
 The `:source_type` option specifies the source association type for a `has_one :through` association that proceeds through a polymorphic association.
 
+```ruby
+class Jpeg < ApplicationRecord; end
+class Png < ApplicationRecord; end
+
+class Profile < ApplicationRecord
+  has_one :picture, polymorphic: true
+end
+
+class User < ApplicationRecord
+  has_one :profile
+  has_one :jpeg, through: :profile, source: :picture, source_type: "Jpeg"
```

Comment:
FWIW, I do not love this example, but really struggled to come up with one for a polymorphic has_one :through. Will gladly take suggestions for a better one!

Reply:
Can we combine the source and source_type examples together? Lots of repetition in both.